src/lib/parser/resolveCircularRef.ts

```
@@ -48,13 +49,26 @@
   while (ancestor) {
-    if (ancestor.value === value) {
+    const ancestorValue = ancestor.value
+    const ancestorTarget =
+      ancestorValue && (ancestorValue as any)[originSymbol]
+        ? (ancestorValue as any)[originSymbol]
+        : ancestorValue
+        
+    if (ancestorTarget === target) {
       return buildReferencePath(ancestor)
     }
     ancestor = ancestor.parent
```

Biome (1.9.4)

[error] Reassigning a function parameter is confusing.

The parameter is declared here:

Use a local variable instead.

(lint/style/noParameterAssign)